bundles/org.openhab.binding.govee/src/main/java/org/openhab/binding/govee/internal/CommunicationManager.java

```diff
@@ -125,24 +144,27 @@ public void runDiscoveryForInterface(NetworkInterface intf, DiscoveryResultRecei
                     activeReceiver.setDiscoveryResultsReceiver(receiver);
                 }
 
-                final InetAddress broadcastAddress = InetAddress.getByName(DISCOVERY_MULTICAST_ADDRESS);
-                final InetSocketAddress socketAddress = new InetSocketAddress(broadcastAddress, RESPONSE_PORT);
-                final Instant discoveryStartTime = Instant.now();
-                final Instant discoveryEndTime = discoveryStartTime.plusSeconds(INTERFACE_TIMEOUT_SEC);
-
-                try (MulticastSocket sendSocket = new MulticastSocket(socketAddress)) {
-                    sendSocket.setSoTimeout(INTERFACE_TIMEOUT_SEC * 1000);
-                    sendSocket.setReuseAddress(true);
-                    sendSocket.setBroadcast(true);
-                    sendSocket.setTimeToLive(2);
-                    sendSocket.joinGroup(new InetSocketAddress(broadcastAddress, RESPONSE_PORT), intf);
+                final Instant discoveryEndTime = Instant.now().plusSeconds(INTERFACE_TIMEOUT_SEC);
 
-                    byte[] requestData = DISCOVER_REQUEST.getBytes();
-
-                    DatagramPacket request = new DatagramPacket(requestData, requestData.length, broadcastAddress,
-                            DISCOVERY_PORT);
-                    sendSocket.send(request);
-                }
+                Collections.list(intf.getInetAddresses()).stream().filter(address -> address instanceof Inet4Address)
+                        .map(address -> address.getHostAddress()).forEach(ipv4Address -> {
+                            try (DatagramChannel channel = (DatagramChannel) DatagramChannel
+                                    .open(StandardProtocolFamily.INET)
+                                    .setOption(StandardSocketOptions.SO_REUSEADDR, true)
+                                    .setOption(StandardSocketOptions.IP_MULTICAST_TTL, 64)
+                                    .setOption(StandardSocketOptions.IP_MULTICAST_IF, intf)
+                                    .bind(new InetSocketAddress(ipv4Address, DISCOVERY_PORT))
+                                    .configureBlocking(false)) {
+                                logger.trace("Datagram channel bound to {}:{} on {}", ipv4Address, DISCOVERY_PORT,
+                                        intf.getDisplayName());
+                                channel.send(ByteBuffer.wrap(DISCOVER_REQUEST.getBytes()),
+                                        new InetSocketAddress(DISCOVERY_MULTICAST_ADDRESS, DISCOVERY_PORT));
```

Comment:
Minor: can create this outside of the loop (or have it as constant right away).

Reply:
Done.